pandas/core/apply.py

```diff
@@ -601,34 +601,32 @@ def normalize_dictlike_arg(
             func = new_func
         return func
 
-    def _try_aggregate_string_function(self, obj, arg: str, *args, **kwargs):
+    def _apply_str(self, obj, arg: str, *args, **kwargs):
         """
         if arg is a string, then try to operate on it:
-        - try to find a function (or attribute) on ourselves
+        - try to find a function (or attribute) on obj
         - try to find a numpy function
         - raise
         """
         assert isinstance(arg, str)
 
-        f = getattr(obj, arg, None)
-        if f is not None:
+        if hasattr(obj, arg):
+            f = getattr(obj, arg)
             if callable(f):
                 return f(*args, **kwargs)
 
-            # people may try to aggregate on a non-callable attribute
+            # people may aggregate on a non-callable attribute
             # but don't let them think they can pass args to it
             assert len(args) == 0
             assert len([kwarg for kwarg in kwargs if kwarg not in ["axis"]]) == 0
```

Comment:
I haven't looked into this, but at least at the surface it seems like users may be able to generate an AssertionError here? Perhaps these should be changed to ValueErrors (also okay to not tackle here).

Reply:
Confirmed:

```
ser = pd.Series([1, 1, 2])
print(ser.agg('size', a=1))
# File "pandas/pandas/core/apply.py", line 621, in _try_aggregate_string_function
#     assert len([kwarg for kwarg in kwargs if kwarg not in ["axis"]]) == 0
# AssertionError
```

Reply:
Yeah, I though about fixing that, but then it wouldn't be just a clean-up. I can do it in a follow-up?